app/models/manageiq/providers/autosde/inventory/collector/target_collection.rb

```diff
@@ -8,49 +8,62 @@ def initialize(_manager, _target)
   def physical_storages
     return [] if references(:physical_storages).blank?
 
-    super
+    @physical_storages ||= references(:physical_storages).map do |ems_ref|
+      @manager.autosde_client.StorageSystemApi.storage_systems_get.map { |s| s if s.uuid == ems_ref }.compact.first
```

Comment:
Also this looks like it will do an API call per target, is there a way to issue a `GET` on a single resource id (or collection of resource ids) without returning the whole collection?

If you do have to do the whole collection then best to only do that once and then filter down the refs like:
```suggestion
    @physical_storages ||= @manager.autosde_client.StorageSystemApi.storage_systems_get.select { |s| references(:physical_storages).include?(s.uuid) }
```